### PR TITLE
plasma support

### PR DESCRIPTION
**Pull request by joco-nz**
Core code needed to support plasma UI build
Errors/quirks with fixes found along the way.

### CODE REVIEW

File: qtpyvcp/actions/program_actions.py [hunk 1]
```diff
@@ -503,8 +503,11 @@ def _optional_stop_bindOk(widget):
 
 def openFilterProgram(infile, prog_name):
```
Comment (silopolis) [inline qtpyvcp/actions/program_actions.py:504]: Hey @joco-nz !

That's a line comment.

File: qtpyvcp/actions/program_actions.py [hunk 2]
```diff
@@ -503,8 +503,11 @@ def _optional_stop_bindOk(widget):
...
     temp_dir = _mktemp()
-    outfile = os.path.join(temp_dir, os.path.basename(infile) + ".ngc")
-    FilterProgram(prog_name, infile, outfile, lambda r: r or _loadFilterResult(outfile))
+    outfile = os.path.join(temp_dir, os.path.basename(infile))
+    #FilterProgram(prog_name, infile, outfile, lambda r: r or _loadFilterResult(outfile))
+    FilterProgram(prog_name, infile, outfile, None)
+    CMD.program_open(outfile)
+    LOG.debug('Linuxcnc Command - program_open')
```
Comment (silopolis) [inline qtpyvcp/actions/program_actions.py:510]: And a multiple lines one :)

**Review by joco-nz — COMMENTED** [review]
**program_actions.py**
Fixes to correctly process the piped data stream from a filter program and load the results.
Multiple tests showed these updates were needed to support correct filter program processing.


**plasma_processes.py**
Database definition and supporting methods.


**status.py**
Added in extra protective code and handling for gantry machines that have y and y1 axis.  Not having this fix was causing issues on my gantry test sim.


**plasma_gcode_preprocessor.py**
Core processor file used as a filter for turning a correctly formed code file into something usable for a plasma.


**plasma_tooldbpipe.py**
Allows the tool number to reference into a cut chart entry.


**vk_backplot.py**
Added in protective code due to errors being through from the monokrom UI.


**hal_slider.py**
Allows a double click to set value to 100%


**action_slider.py**
Allows a double click to set value to 100%


**setting_slider.py**
Allows a double click to set value to 100%
Added editing ended signal so as to get expected behaviours on focus end.


**setup.py**
Added in sqlalchemy dependency.
Added in new entry points.

File: qtpyvcp/actions/program_actions.py [hunk 3]
```diff
@@ -503,8 +503,11 @@ def _optional_stop_bindOk(widget):
...
 
 def _loadFilterResult(fname):
```
Comment (silopolis) [inline qtpyvcp/actions/program_actions.py:512]: Add single line comment

File: qtpyvcp/actions/program_actions.py [hunk 4]
```diff
@@ -529,19 +532,20 @@ def __init__(self, prog_name, infile, outfile, callback=None):
 
         env = dict(os.environ)
         env['AXIS_PROGRESS_BAR'] = '1'
-        self.p = subprocess.Popen(["sh", "-c", "%s '%s'" % (prog_name, infile)],
+        self.p = subprocess.run(["sh", "-c", "%s '%s'" % (prog_name, infile)],
                                   stdin=subprocess.PIPE,
```
Comment (silopolis) [inline qtpyvcp/actions/program_actions.py:536]: Add multiline comment

**Review by TurBoss — APPROVED** [review]
wow! awesome work @joco-nz

**Comment by silopolis** [comment]
That's awesome!

Great work @joco-nz 👍

**Pull request merged**